shell/platform/linux/fl_engine_impl_egl.cc

```diff
@@ -0,0 +1,166 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "fl_engine_impl_egl.h"
+
+#include <EGL/egl.h>
+#include <EGL/eglext.h>
+#include <GLES2/gl2.h>
+
+#include "flutter/shell/platform/embedder/embedder.h"
+
+typedef struct {
+  EGLint egl_major;
+  EGLint egl_minor;
+
+  EGLDisplay egl_display;
+  EGLSurface egl_surface;
+  EGLContext egl_context;
+} FlEngineImplEGLPrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE(FlEngineImplEGL,
+                           fl_engine_impl_egl,
+                           fl_engine_impl_get_type())
+
+static gboolean fl_engine_impl_egl_start(FlEngineImpl* engine, GError** error) {
+  FlEngineImplEGL* self = FL_ENGINE_IMPL_EGL(engine);
+  FlEngineImplEGLPrivate* priv = static_cast<FlEngineImplEGLPrivate*>(
+      fl_engine_impl_egl_get_instance_private(self));
+
+  /* Note that we don't provide the XDisplay from GTK, this would make both
+   * GTK and EGL share the same X connection and this would crash when used by
```

Comment:
This is because inside eglGetDisplay it's making an X connection (or Wayland connection or whatever else). I'll update the comment to make this clearer as it was written when this code was in the same file as the GTK code.